proxy/hdrs/URL.cc

```diff
@@ -1264,7 +1287,7 @@ url_parse_internet(HdrHeap *heap, URLImpl *url, char const **start, char const *
       last_colon = 0; // prevent port setting.
     }
   }
-  if (host._size)
+  if (host._size && validate_host_name(host))
```

Comment:
I think here the condition should be to check `host._size` first and then for `validate_host_hdr` either set the host (if success) or generate an error (if false). It is acceptable to not have a host at all in the URL, but it is unacceptable (an error) if there is a host and it's malformed.